### DISCUSSION TITLE
Exporting data, t, x, y

### DISCUSSION THREAD

**Discussion by hesmalek** [zwicker-group/py-pde#236] [Q&A]
Hi,
How can I export the raw data (grid, Y and time) easily to e.g. CSV file? 

for example for Stochastic simulation code: 

--

from pde import KPZInterfacePDE, MemoryStorage, ScalarField, UnitGrid, plot_kymograph

grid = UnitGrid([64])  # generate grid
state = ScalarField.random_harmonic(grid)  # generate initial condition

eq = KPZInterfacePDE(noise=1)  # define the SDE
storage = MemoryStorage()
eq.solve(state, t_range=10, dt=0.01, tracker=storage.tracker(0.5))
plot_kymograph(storage)

--

Thanks

**Comment by david-zwicker** [comment]
You can replace the `MemoryStorage` by a `FileStorage`, which writes all this information to an hdf file, which can be read by many programs. CSV files are not flexible enough to store data in all cases, but you can of course access all raw data from python and write your own format. If you find a solution that is general enough (and thus could also be useful for other people), I'm happy to look at and accept pull request!
